hibernate-reactive-core/src/main/java/org/hibernate/reactive/event/impl/DefaultReactiveDeleteEventListener.java

```diff
@@ -337,59 +337,56 @@ private void disallowDeletionOfDetached(DeleteEvent event) {
 		persistenceContext.setEntryStatus( entityEntry, Status.DELETED );
 		final EntityKey key = session.generateEntityKey( entityEntry.getId(), persister );
 
-		CompletionStage<?> beforeDelete = cascadeBeforeDelete( session, persister, entity, entityEntry, transientEntities );
-
-		CompletionStage<Void> nullifyAndAction = new ForeignKeys.Nullifier(
-				entity,
-				true,
-				false,
-				session,
-				persister
-		).nullifyTransientReferences( entityEntry.getDeletedState() )
-				.thenAccept( v -> {
-					new Nullability( session ).checkNullability(
-							entityEntry.getDeletedState(),
-							persister,
-							Nullability.NullabilityCheckType.DELETE
-					);
-					persistenceContext.registerNullifiableEntityKey( key );
-
-					ReactiveActionQueue actionQueue = actionQueue( session );
-
-					if ( isOrphanRemovalBeforeUpdates ) {
-						// TODO: The removeOrphan concept is a temporary "hack" for HHH-6484.  This should be removed once action/task
-						// ordering is improved.
-						actionQueue.addAction(
-								new ReactiveOrphanRemovalAction(
-										entityEntry.getId(),
-										deletedState,
-										version,
-										entity,
-										persister,
-										isCascadeDeleteEnabled,
-										session
-								)
-						);
-					}
-					else {
-						// Ensures that containing deletions happen before sub-deletions
-						actionQueue.addAction(
-								new ReactiveEntityDeleteAction(
-										entityEntry.getId(),
-										deletedState,
-										version,
-										entity,
+		return cascadeBeforeDelete( session, persister, entity, entityEntry, transientEntities )
+				.thenCompose( beforeDelete -> new ForeignKeys.Nullifier(
+							entity,
+							true,
+							false,
+							session,
+							persister
+					).nullifyTransientReferences( entityEntry.getDeletedState() )
+							.thenAccept( v -> {
+								new Nullability( session ).checkNullability(
+										entityEntry.getDeletedState(),
 										persister,
-										isCascadeDeleteEnabled,
-										session
-								)
-						);
-					}
-				} );
-
-		CompletionStage<Void> afterDelete = cascadeAfterDelete( session, persister, entity, transientEntities );
```

Comment:
Hrm. @Davide I actually had this being called outside of the `thenCompose()` callback intentionally.

So this change actually changes the behavior of the code somewhat.

It probably doesn't break anything but I'm still not sure I like it.